src/storage.js

```diff
@@ -23,4 +25,39 @@ module.exports = {
   getFeeds: () => redis.smembers(FEEDS),
 
   getFeed: feedID => redis.hgetall(feedID),
+
+  addPost: async post => {
+    await redis
+      .multi()
+      .hmset(
+        // using guid as keys as it is unique to posts
+        post.guid,
+        'author',
+        post.author,
+        'title',
+        post.title,
+        'link',
+        post.link,
+        'content',
+        post.content,
+        'text',
+        post.text,
+        'updated',
+        post.updated,
+        'published',
+        post.published,
+        'url',
+        post.url,
+        'site',
+        post.site
+      )
+      .zadd(POSTS, new Date(post.published).getTime(), post.guid) // sort set by published date as scores
+      .exec();
+  },
+
+  getPosts: (startDate, endDate) =>
```

Comment:
Not sure about startDate and endDate. You could just pass it as Date parameter.

Reply:
I agree with @mskuybeda, I think we could pass a Date and get everything between that date and [the current date](https://javascript.info/date#creation).